client/views/view_cities.cpp

```diff
@@ -732,6 +705,48 @@ void city_widget::display_list_menu(const QPoint)
   list_menu->popup(QCursor::pos());
 }
 
+/**
+   Sell building for all selected cities after asking for
+   confirmation.
+*/
+void city_widget::sell(const struct impr_type *building)
+{
+  fc_assert_ret(building != nullptr);
+
+  Impr_type_id impr_id = improvement_number(building);
+  if (nullptr == improvement_by_number(impr_id)) {
+    return;
+  }
+
+  hud_message_box *ask = new hud_message_box(king()->central_wdg);
+  QString imprname = improvement_name_translation(building);
+  QString buf;
+  if (selected_cities.size() == 1) {
+    buf = QString(_("Are you sure you want to sell the %1?")).arg(imprname);
+  } else {
+    buf = QString(_("Do you really wish to sell every %1?")).arg(imprname);
```

Comment:
I think "every" can be misinterpreted as "every city you own".
```suggestion
    buf = QString(_("Do you really wish to sell all selected %1?")).arg(imprname);
```